app/javascript/controllers/claims_support_filter_search_controller.js

```diff
@@ -0,0 +1,50 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="claims-support-filter-search"
+export default class extends Controller {
+  static targets = [ "schoolInput", "schoolList", "providerInput", "providerList" ]
+
+  connect() {
+    this.schoolInputTarget.addEventListener('input', this.searchSchool.bind(this))
```

Comment:
There's a _memory leak_ with this code here, it won't be an issue for our app, as our app doesn't use Turbo and performs whole-page reloads, but nevertheless.

To register event listeners with Stimulus that are also cleaned up automatically, use the following approach: https://stimulus.hotwired.dev/reference/actions

In this case, it would be:

```erb
<%= form.govuk_text_field(
              :search_school,
              type: :search,
              data: { claims_support_filter_search_target: "schoolInput", action: "claims-support-filter-search#searchSchool" },
              label: { text: t("claims.support.claims.index.school"), size: "s" },
            ) %>
```

The default "event" type for `input` tags, is `input`.

![CleanShot 2024-04-12 at 10 32 43](https://github.com/DFE-Digital/itt-mentor-services/assets/4231530/016575c5-31a6-4f27-bcae-c236f0f4afec)